app/mock_svc.py

```diff
@@ -29,6 +30,16 @@ def __init__(self, services):
                                 response=self.agent_svc.encode_string(p['response']))
                 await self.data_svc.create('sim_response', response)
 
+    async def set_agent_listing(self, agents):
+        self.agents = agents
```

Comment:
can we pass the 'all_agents' from the hook.py into the MockService constructor (__init__)? This would negate this function and remove the 'code smell' of setting an init variable outside of init.

Reply:
I can pull that off. I'd kept it out originally because inits seem to usually be restricted to just services, but it really doesn't make sense to have a separate function for this in the end.